### DISCUSSION TITLE
How to have custom API response

### DISCUSSION THREAD

**Discussion by emaddeldin** [encode/django-rest-framework#8535] [Question & Answer]
So it's been pretty easy to serialize a table and return its data as a JSON response using the REST framework. 
However, I wish to return custom data not related to any table or database for specific endpoints in my Django application.
Any help on this would be very much appreciated 🙏🏼

**Comment by Azanul** [comment]
Do you mean something like this:
```
from rest_framework.views import APIView
from rest_framework.response import Response


class MyAPIView(APIView):
        def get(self, request, *args, **kwargs):
                return Response({"data": "Custom Data"}, status=200)
```

**Reply by emaddeldin** [reply]
yes exactly like this, can you elaborate on how its done

**Reply by Azanul** [reply]
I assuming you're already familiar with ```APIView```.
```def get``` defines the function to be run on a get request.
In ```Response```  ([ref](https://www.django-rest-framework.org/api-guide/responses/#response)) mention the dictionary that'll be sent as the requested content type (Json, xml, etc) and in status the status code (200 for OK, 500 for Internal Server error, [etc](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)
